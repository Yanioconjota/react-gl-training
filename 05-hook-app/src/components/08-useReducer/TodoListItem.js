import React from "react";
import PropTypes from "prop-types";

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {
  return (
    <li
      key={todo.id}
      className="list-group-item d-flex justify-content-between align-items-center">
      <p
        className={`${todo.done && "complete"}`}
        onClick={() => handleToggle(todo.id)}>
        {index + 1}. {todo.desc}
      </p>
      <div className="btn-group">
        <button onClick={() => handleDelete(todo.id)} className="btn btn-danger">
          ✖️
        </button>
      </div>
    </li>
  );
};

TodoListItem.propTypes = {
  todo: PropTypes.object,
  index: PropTypes.number, 
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func,
};

export default TodoListItem;
