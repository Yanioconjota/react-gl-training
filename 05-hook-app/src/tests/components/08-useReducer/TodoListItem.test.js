import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe('pruebas con TodoListItem', () => {
  const [ todo ] = demoTodos;
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const index = 0;
  const wrapper = shallow(
    <TodoListItem
      todo={todo}
      index={index}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('el componente debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe llamar la función borrar', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    //espera que se llama handleDelete(todo.id) donde 1 es el todo.id del todo que se quiere borrar
    expect(handleDelete).toHaveBeenCalledWith(todo.id);
  });

  test('debe llamar la función toggle', () => {
    const todoItem = wrapper.find('p');
    todoItem.simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(todo.id);
  });

  test('debe mostrar el texto correctamente', () => {
    const todoItem = wrapper.find("p");
    console.log(todoItem.text());
    expect(todoItem.text().trim()).toBe(`${index + 1}. ${todo.desc}`);
  });

  test('debe tener la clase complete si el todo.done = true', () => {
    todo.done = true;  
    const wrapper = shallow(
      <TodoListItem todo={todo}/>
    );
    expect(wrapper.find('p').hasClass('complete')).toBe(true);
  });
  
});
