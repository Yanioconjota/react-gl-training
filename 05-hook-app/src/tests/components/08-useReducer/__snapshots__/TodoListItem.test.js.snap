// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con TodoListItem el componente debe mostrarse correctamente 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Aprender react
  </p>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      ✖️
    </button>
  </div>
</li>
`;
